static/examples/hashing-tool/index.html

```diff
@@ -0,0 +1,167 @@
+  </head>
+  <script>
+    console.log('Initializing example.');
+    
+
+    
+
+    async function updateGuiElements(input) {
+      console.log('Updating displayed values.');
+      const input_type = document.querySelector('input[name="toggle_input_type"]:checked').value;
+      let normalizedInput = input;
+      if (input_type === "email") {
+        normalizedInput = window.__uid2Helper.normalizeEmail(input);
+        $('#normalization').text(normalizedInput);
+      }
+      
+      if (normalizedInput) {
+        const hashedValue = await window.__uid2Helper.hashIdentifier(normalizedInput);
+        $('#hashed').text(hashedValue);
+        const encodedValue = await window.__uid2Helper.hashAndEncodeIdentifier(normalizedInput);
+        $('#base64_encoded').text(encodedValue);  
+      }
+      else {
+        $('#normalization').text(' ');
+        $('#hashed').text(' ');
+        $('#base64_encoded').text(' ');
+        $('#alert_box_email').show();
+      }
+    }
+
+    function clearElements() {
+      document.querySelector('.input_value').value = "";
+      $('#normalization').text(' ');
+      $('#hashed').text(' ');
+      $('#base64_encoded').text(' ');
+      $('.alert').hide();
+    }
+
+    function onDocumentReady() {
+        console.log('Setting up interface handlers.');
+        document.querySelector('.input_value').placeholder = "Enter an email address";
+        $('#enter').click(handleEnter);
+        $('#clear_values_form').click(clearElements);
+        $('.alert').hide();
+        $('#clear_values_form').show();
+        $('#normalization').text(' ');
+        $('#hashed').text(' ');
+        $('#base64_encoded').text(" ");
+    }
+
+    function handleRadioClick(radioButton) {
+      clearElements();
+      const input_type = radioButton.value;
+      let placeholder = "";
+      if (input_type == "email") {
+        placeholder = "Enter an email address";
+        $('#normalization').show();
+        $('#normalization_label').show();
+      } 
+      else if (input_type == "mobile") {
+        placeholder = "Enter a phone number";
+        $('#normalization').hide();
+        $('#normalization_label').hide();
+      } 
+      document.querySelector('.input_value').placeholder = placeholder;
+    }
+
+    async function handleEnter() {
+        const input_value = $('#input_value').val();
+        const input_type = document.querySelector('input[name="toggle_input_type"]:checked').value;
+        if (input_type == "mobile") {
+          const isNormalizedPhone = window.__uid2Helper.isNormalizedPhone(input_value)
+          if (!isNormalizedPhone) {
+            $('#hashed').text(" ");
+            $('#base64_encoded').text(" ");
+            document.querySelector('.alert').innerText = "Phone number format is invalid or is not normalized."
+            $('#alert_box_mobile').show();
+            return;
+          }
+        }
+        updateGuiElements(input_value);
```

Comment:
Why don't we have both the email and phone number logic in the same place?

Reply:
It was because the logic is a bit different since we normalize emails but not phone numbers so there is an extra step.  But I cleaned this up a bit so there is email & phone number logic in `handleEnter()` and something more generalized that applies to both input types in `updateGuiElements `